src/fortran/curve.f90

```diff
@@ -93,6 +93,63 @@ subroutine evaluate_curve_barycentric( &
             lambda2_pow(i) * lambda2(i) * nodes(:, num_nodes))
     end forall
 
+  end subroutine evaluate_curve_vs
+
+  subroutine evaluate_curve_de_casteljau( &
+       num_nodes, dimension_, nodes, num_vals, lambda1, lambda2, evaluated)
+
+    ! NOTE: This is evaluate_multi_de_casteljau for a Bezier curve.
+
+    integer(c_int), intent(in) :: num_nodes, dimension_
+    real(c_double), intent(in) :: nodes(dimension_, num_nodes)
+    integer(c_int), intent(in) :: num_vals
+    real(c_double), intent(in) :: lambda1(num_vals)
+    real(c_double), intent(in) :: lambda2(num_vals)
+    real(c_double), intent(out) :: evaluated(dimension_, num_vals)
+    ! Variables outside of signature.
+    integer(c_int) :: i
+    real(c_double) :: lambda1_wide(dimension_, num_vals, num_nodes - 1)
+    real(c_double) :: lambda2_wide(dimension_, num_vals, num_nodes - 1)
+    real(c_double) :: workspace(dimension_, num_vals, num_nodes - 1)
+
+    forall (i = 1:num_vals)
+       lambda1_wide(:, i, :) = lambda1(i)
+       lambda2_wide(:, i, :) = lambda2(i)
+       workspace(:, i, :) = ( &
+            lambda1(i) * nodes(:, :num_nodes - 1) + lambda2(i) * nodes(:, 2:))
+    end forall
+
+    do i = num_nodes - 2, 1, -1
+       workspace(:, :, :i) = ( &
+            lambda1_wide(:, :, :i) * workspace(:, :, :i) + &
+            lambda2_wide(:, :, :i) * workspace(:, :, 2:(i+1)))
```

Comment:
I'd like to toy with using `forall` here instead of making the quite large `lambda1_wide` / `lambda2_wide` variables.